src/startup.rs

```diff
@@ -36,13 +38,17 @@ pub struct HttpServer {
 }
 
 impl HttpServer {
-    pub async fn new(config: &Config, state: AppState) -> color_eyre::Result<Self> {
+    pub async fn new(
+        config: &Config,
+        state: AppState,
+        metrics_handle: Option<PrometheusHandle>,
+    ) -> color_eyre::Result<Self> {
```

Comment:
I do not consider sustainable passing handlers to this constructor.
Implement all the loading logic as a function that you import into this module and simply call.
```suggestion
    pub async fn new(
        config: &Config,
        state: AppState
    ) -> color_eyre::Result<Self> {
```